### PR TITLE
Small refactoring, Chartist update and a fix

### PR DESCRIPTION
**Pull request by gionkunz**
Hi Will

I'm currently working on a project where I use your Angular module :-) I just needed to update to the latest Chartist version and I've also found a small bug. When the chart type changed together with the data, then your module was first doing an update and directly after that re-created the chart which caused some issues because of the async chart creation nature of Chartist. I've also simplified the watch functionality as I believe checking for chartType changes is important but if something else caused the watch to trigger we could simply assume it was either the options or data and just do a update in that case.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,7 +1,8 @@
 root = true
 
 [*]
-indent_style = tab
+indent_style = space
```
Comment (gionkunz) [inline .editorconfig]: I've just added this according to your current style I found in the files

File: dist/angular-chartist.js [hunk 2]
```diff
@@ -22,17 +22,15 @@
         this.responsiveOptions = $scope.responsiveOptions() || null;
     }
 
+    AngularChartistCtrl.$inject = ['$scope'];
```
Comment (gionkunz) [inline dist/angular-chartist.js]: I believe this is more maintainable over the `[dep, dep, dep, function]` notation in injectable targets... maybe only my preference?

Comment (paradox41) [inline dist/angular-chartist.js]: Actually that is great! I have been meaning to switch to this style over the dep array.

File: .editorconfig (hunk 1 above)
Comment (paradox41) [inline .editorconfig]: Thanks! That is right.

**Comment by paradox41** [comment]
LGTM. I'll go ahead and get this released today.

**Pull request merged**